src/types/account-manager.ts

```diff
@@ -0,0 +1,253 @@
+    this._accounts[sender] = { addr: sender, signer }
+    return this
+  }
+
+  /**
+   * Returns the `TransactionSigner` for the given sender address.
+   *
+   * If no signer has been registered for that address then the default signer is used if registered.
+   *
+   * @param sender The sender address
+   * @returns The `TransactionSigner` or throws an error if not found
+   */
+  public getSigner(sender: string): algosdk.TransactionSigner {
+    const signer = this._accounts[sender]?.signer ?? this._defaultSigner
+    if (!signer) throw new Error(`No signer found for address ${sender}`)
+    return signer
+  }
+
+  /**
+   * Returns the `TransactionSignerAccount` for the given sender address.
+   * @param sender The sender address
+   * @returns The `TransactionSignerAccount` or throws an error if not found
+   */
+  public getAccount(sender: string): TransactionSignerAccount {
+    const account = this._accounts[sender]
+    if (!account) throw new Error(`No signer found for address ${sender}`)
+    return account
+  }
+
+  /**
+   * Returns the given sender account's current status, balance and spendable amounts.
+   *
+   * @example
+   * ```typescript
+   * const address = "XBYLS2E6YI6XXL5BWCAMOA4GTWHXWENZMX5UHXMRNWWUQ7BXCY5WC5TEPA";
+   * const accountInfo = await account.getInformation(address);
+   * ```
+   *
+   * [Response data schema details](https://developer.algorand.org/docs/rest-apis/algod/#get-v2accountsaddress)
+   * @param sender The address of the sender/account to look up
+   * @returns The account information
+   */
+  public async getInformation(sender: string | TransactionSignerAccount): Promise<AccountInformation> {
+    return getAccountInformation(sender, this._clientManager.algod)
+  }
+
+  public async getAssetInformation(sender: string | TransactionSignerAccount, assetId: bigint) {
+    return getAccountAssetInformation(sender, assetId, this._clientManager.algod)
+  }
+
+  /**
+   * Tracks and returns an Algorand account with secret key loaded (i.e. that can sign transactions) by taking the mnemonic secret.
+   *
+   * @example
+   * ```typescript
+   * const account = await account.fromMnemonic("mnemonic secret ...")
+   * const rekeyedAccount = await account.fromMnemonic("mnemonic secret ...", "SENDERADDRESS...")
+   * ```
+   * @param mnemonicSecret The mnemonic secret representing the private key of an account; **Note: Be careful how the mnemonic is handled**,
+   *  never commit it into source control and ideally load it from the environment (ideally via a secret storage service) rather than the file system.
+   * @param sender The optional sender address to use this signer for (aka a rekeyed account)
+   * @returns The account
+   */
+  public fromMnemonic = (mnemonicSecret: string, sender?: string) => {
+    const account = mnemonicAccount(mnemonicSecret)
+    return this.signerAccount(sender ? rekeyedAccount(account, sender) : account)
```

Comment:
I don't think `this` will work as `this` won't be bound to the class instance in an arrow function.

Reply:
Hmmm, apparently it does work but only because the assignment is hoisted to the constructor. TIL!